src/components/core/geoChart/geoChartDataSource.ts

```diff
@@ -104,6 +127,86 @@ function transformPushpinDataSource(
     return features;
 }
 
+function getDrillableProps(dataSourceProps: IGeoDataSourceProps, dataIndex: number): IDrillableProps {
+    const { drillableItems, drillConfig, execution, geoData } = dataSourceProps;
+    const { executionResponse } = execution;
+    const { dimensions } = executionResponse;
+
+    const measureGroup = findMeasureGroupInDimensions(dimensions);
+    const measureHeaders: Execution.IMeasureHeaderItem[] = get(measureGroup, "items", []).slice(0, 2);
+
+    const { locationAttribute, segmentByAttribute } = findGeoAttributesInDimension(execution, geoData);
+    const {
+        attributeHeader: locationAttributeHeader,
+        attributeHeaderItem: locationAttributeHeaderItem,
+    } = getAttributeHeader(locationAttribute, dataIndex);
+    const {
+        attributeHeader: segmentByAttributeHeader,
+        attributeHeaderItem: segmentByAttributeHeaderItem,
+    } = getAttributeHeader(segmentByAttribute, dataIndex);
+
+    // pin is drillable if a drillableItem matches:
+    //   pin's measure,
+    //   pin's location attribute,
+    //   pin's location attribute item,
+    //   pin's segmentBy attribute,
+    //   pin's segmentBy attribute item,
+    const drillableHooks: IMappingHeader[] = without(
+        [
+            ...measureHeaders,
+            locationAttributeHeaderItem,
+            locationAttributeHeader,
+            segmentByAttributeHeaderItem,
+            segmentByAttributeHeader,
+        ],
+        undefined,
+    );
+
+    const drilldown: boolean = drillableHooks.some(
+        (drillableHook: IMappingHeader): boolean =>
+            isSomeHeaderPredicateMatched(drillableItems, drillableHook, drillConfig.afm, executionResponse),
+    );
+
+    const drillableProps: IDrillableProps = {
+        drilldown,
+    };
+
+    if (drilldown) {
+        const drillItems: IMappingHeader[] = without(
```

Comment:
Is this same to `drillableHooks`? If so, we should reuse

Reply:
Good catching.
Thanks

Reply:
updated